contractcourt/chain_watcher.go

```diff
@@ -974,24 +977,62 @@ func (c *chainWatcher) handleUnknownRemoteState(
 // return zero. This is possible as our output may have been trimmed due to
 // being dust.
 func (c *chainWatcher) toSelfAmount(tx *wire.MsgTx) btcutil.Amount {
-	var selfAmt btcutil.Amount
-	for _, txOut := range tx.TxOut {
+	// There are two main cases we have to handle here. First, in the coop
+	// close case we will always have saved the delivery address we used
+	// whether it was from the upfront shutdown, from the delivery address
+	// requested at close time, or even an automatically generated one. All
+	// coop-close cases can be identified in the following manner:
+	shutdown, _ := c.cfg.chanState.ShutdownInfo()
```

Comment:
so we do not handle DB error, here?

Reply:
Since this function doesn't return an error, if we can't reach the shutdown info for whatever reason we treat it as non-existent. Why it failed is inconsequential.

Reply:
Oh okay, I was just thinking that maybe if there is something wrong with the DB we can catch it as soon as we encounter it here.

Reply:
If this function already had an error in its signature then I'd probably do that. But this would cause a currently non-erroring function to return an error which I'd like to avoid. Swallowing errors *can* be dangerous but in this case all it does is affect the numbers we report and I don't think that's especially problematic. If it comes out to result in bad UX's we can come back and address it but for now I'd recommend leaving it as is.